discussions/micro-frontends/base-paths.md

```diff
@@ -0,0 +1,150 @@
+        ^^^^^^^^^^^^^
+```
+
+(Note the missing `header` from the domain name.)
+
+What we need is to ensure that the path to the logo will result in the correct asset being accessed.
+There are two general approaches:
+
+1. Always use full absolute URLs for all assets, so that asset requests go directly to the URL hosting the fragment (and its assets).
+2. Delegate requests for assets through the Worker that is hosting the container application.
+
+### Absolute asset URLs
+
+On the face of it this seems like a reasonable approach, and it is not that different to traditional approaches where assets are uploaded to a separate CDN.
+For this to work, it is important that each fragment has the appropriate CORS configuration, to allow cross-origin requests from the application domain to the fragment domain.
+It is not enough to hard-code absolute paths into templates and code since this restricts deploying to different domains, which is especially relevant when the fragment is in development.
+The challenge therefore is to ensure that whatever tooling is used (e.g. meta-framework) it can be configured to render HTML and client-side code with the appropriate absolute path to the fragment.
+In practice this is not always straightforward, since many tools will only provide relative paths to assets rather than full absolute URLs.
+
+### Delegating asset requests
+
+Rather than asking the browser to make requests for assets directly to the fragment we could delegate the request through the container application's Worker.
+This is particularly useful if, as in our cloud-gallery demo, the fragments are connected to the container application via "service bindings".
+This avoids making real HTTP fetch requests between Workers.
+For delegation to work, the paths for fragment assets must include some information that the container application's Worker can use to select the fragment to which the request should be delegated.
+For example, if an asset such as `/logo.gif` needs to be served by the "header" fragment, then the request from the browser could be `/_fragment/header/logo.gif`.
+This would result in a request to https://cloud-gallery.web-experiments.workers.dev/_fragment/header/cf-logo.png`.
+The container application Worker would the parse the path to identify that this is an asset request for the `header` fragment and forward the request through to the `header` fragment Worker, having stripped the `_fragment/header/` prefix.
+
+## Configuring asset paths
+
+Asset paths are only problematic if they are requested in the context of the container application rather than the micro-frontend.
+For example the source of an image tag ( <img src="/images/logo.gif"> ) will have the wrong context if it was generated by a fragment (it would need to be something like <img src="https://cdn.mydomain.com/images/logo.gif"> or <img src="/_fragment/header/images/logo.gif"> ).
+The primary places this can happen are:
```

Comment:
If the `url()` is inside a stylesheet file, then it is relative to that file. From MDN:

> Relative URLs, if used, are relative to the URL of the stylesheet (not to the URL of the web page).

If the `url()` is an inline style then it will be relative to the containing document, which is a problem, I think.
I'll add another bullet.

Reply:
I see 🙂 

anyways I am a bit unsure if it's all ok with external css files (it may totally be and I'm just confused 😜), if we proxy everything from a gateway worker (thus the browser does not know of the fragment url), are relative paths going to be successfully served? Since the requests will need to pass through the gateway worker they can lose the prefix and the gateway worker can be unable to know to which fragment redirect the requests too?

Reply:
If you have a stylesheet hosted in a fragment a `/styles/x.css`. And it is requested in the application via `/_fragment/foo/styles/x.css`. Then if there is a `url('../images/logo.gif')` reference in the `x.css` stylesheet the browser will make a request to: `/_fragment/foo/images/logo.gif`. Right?

One thing we might need to be careful about is when we do things like:

```
import css from "./styles/x.css";

...

// use the styles in a component
```

I don't know if the build tooling would be able to resolve the fragment prefix...